ciao-image/datastore/metads.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2016 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datastore
+
+import (
+	"github.com/01org/ciao/database"
+	"github.com/golang/glog"
+)
+
+// MetaDs implements the DataStore interface for persistent data
+type MetaDs struct {
+	database.DbProvider
+	DbDir  string
+	DbFile string
+}
+
+//  closeDb is a generic function to close every Db transaction
+func closeDb(metaDs *MetaDs) {
+	_ = metaDs.DbClose()
+}
+
+// Write is the metadata write implementation.
+func (m *MetaDs) Write(i Image) error {
+	defer closeDb(m)
+
+	err := m.DbInit(m.DbDir, m.DbFile)
```

Comment:
Why do we open and close the database on each operation?  Is there any reason why we don't init the database once when we start the image service and close it when image service shutdowns.